### PR TITLE
Allow UA to clamp maxBufferSize to a value between 1 and the value provided by the web page

### PR DESCRIPTION
**Pull request by youennf**
Fixes https://github.com/w3c/mediacapture-transform/issues/95

***
<a href="https://pr-preview.s3.amazonaws.com/youennf/mediacapture-transform/pull/105.html" title="Last updated on Jan 29, 2024, 10:09 AM UTC (0708187)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-transform/105/f57ac6a...youennf:0708187.html" title="Last updated on Jan 29, 2024, 10:09 AM UTC (0708187)">Diff</a>

### CODE REVIEW

**Review by henbos — APPROVED** [review]

File: index.bs [hunk 1]
```diff
@@ -204,7 +212,7 @@ with |processor| as parameter.
 
 The <dfn>handleNewFrame</dfn> algorithm is given a |processor| as input.
 It is defined by running the following steps:
-1. If |processor|.`[[maxBufferSize]]` has a value and |processor|.`[[queue]]` has |processor|.`[[maxBufferSize]]` elements, [=queue/dequeue=] an item from |processor|.`[[queue]]`.
+1. If |processor|.`[[queue]]` has |processor|.`[[maxBufferSize]]` elements, [=queue/dequeue=] an item from |processor|.`[[queue]]`.
```
Comment (henbos) [inline index.bs:215]: You probably need to close the frame that you dequeued

Comment (alvestrand) [reply index.bs:215]: If we close it here, we should also use "dequeue and close" on line 219.

Comment (youennf) [reply index.bs:215]: I'll file an issue about this.
For camera tracks, what we enqueue here is not a WebCodecs VideoFrame object but the necessary information to create one.
For VideoTrackGenerator tracks given to MediaStreamTrackProcessor, we could indeed directly enqueue VideoFrames but we would not expose this VideoFrame object but a clone.

**Comment by youennf** [comment]
Merging this PR given I filed https://github.com/w3c/mediacapture-transform/issues/106 to handle the comment.

**Pull request merged**